### PR TITLE
Unifying c2s metrics

### PR DESCRIPTION
**Pull request by DenysGonchar**
this PR introduces the following changes:
- removed `global.data.xmpp.*.compressed_size` metrics - compression support is removed a while ago.
- removed `global.data.xmpp.*.encrypted_size` metrics - the size of a stanza doesn't change much after encryption, more over, when encryption is done for a batch of stanza then this metric is completely irrelevant.
- improved `global.data.xmpp.*.xml_stanza_size` reporting - this histogram metrics make sense only when reported for a single stanza, this PR ensures that.
- introduced `global.data.xmpp.*.c2s*.raw` spiral metrics for TCP/TLS/BOSH/WebSockets - firstly spiral calculation makes much more sense than histogram in addition to that it's less expensive, in the upcoming PR support for S2S and XMPP components traffic calculation will be added.
- removed support of `mongoose_transport` for bosh socket - that's not needed any more, the same should be done for websockets in the following PR (as well as support of websockets connection for XMPP components)

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/166440) / small_tests / fe0e673cb705b9c3539573dfc49699ab53a58671
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3967/166440/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3967/166440/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2023-02-20_07.55.46/lib.mongooseim.logs/run.2023-02-20_07.55.46/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/166580) / small_tests / 149d420e4eefede5a1fae579ffecaa3a639cc5cb
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3967/166580/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3967/166580/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2023-02-20_20.50.53/lib.mongooseim.logs/run.2023-02-20_20.50.53/suite.log.html)

**Review by NelsonVides — COMMENTED** [review]
I'm generally very happy with all the changes, the only thing is that some are kinda breaking ones so they need documentation and potentially migration, which is something @chrzaszcz is working on. Dunno how do you want to proceed with that but apart from docs, this PR is exactly what I wanted, thank you so much 🎉

**Comment by DenysGonchar** [comment]
> I'm generally very happy with all the changes, the only thing is that some are kinda breaking ones so they need documentation and potentially migration, which is something @chrzaszcz is working on. Dunno how do you want to proceed with that but apart from docs, this PR is exactly what I wanted, thank you so much 🎉

I was going to add documentation in the upcoming PR, when all the metrics are added (incl. s2s and xmpp components connections)

File: big_tests/tests/vcard_simple_SUITE.erl [hunk 1]
```diff
@@ -456,7 +456,7 @@ configure_mod_vcard(Config) ->
 ldap_opts() ->
     LDAPOpts = #{filter => <<"(objectClass=inetOrgPerson)">>,
                  base => <<"ou=Users,dc=esl,dc=com">>,
-                 search_fields => [{"Full Name", "cn"}, {"User", "uid"}],
+                 search_fields => [{<<"Full Name">>, <<"cn">>}, {<<"User">>, <<"uid">>}],
                  vcard_map => [{"FN", "%s", ["cn"]}]},
```
Comment (chrzaszcz) [inline big_tests/tests/vcard_simple_SUITE.erl:460]: Just a note: these should be binaries too.

File: doc/operation-and-maintenance/MongooseIM-metrics.md [hunk 2]
```diff
@@ -163,10 +163,6 @@ Metrics specific to an extension, e.g. Message Archive Management, are described
 | ----------- | ---- | ----------- |
 | `[global, data, xmpp, received, xml_stanza_size]` | histogram | A size (in bytes) of a received stanza after decompression and decryption. |
 | `[global, data, xmpp, sent, xml_stanza_size]` | histogram | A size (in bytes) of a stanza sent to a client socket. |
-| `[global, data, xmpp, received, compressed_size]` | histogram | A size (in bytes) of a received stanza before decompression. |
```
Comment (chrzaszcz) [inline doc/operation-and-maintenance/MongooseIM-metrics.md:166]: Maybe describe all new metrics?

**Review by chrzaszcz — COMMENTED** [review]
Looks good in general, I added a few comments. There is a need to update migration guide for metrics, but we could do it later.

**Review by chrzaszcz — APPROVED** [review]
We agreed with @DenysGonchar that the comments will be addressed in a separate PR.

**Pull request merged**